tests/sign_transaction.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+
+import sys
+import json
+import argparse
+
+from pathlib import Path
+
+from ragger.utils import pack_derivation_path
+from ragger.backend import LedgerCommBackend
+
+
+REPO_ROOT_DIRECTORY = Path(__file__).parent
+EOS_LIB_DIRECTORY = (REPO_ROOT_DIRECTORY / "../tests/functional/apps").resolve().as_posix()
+sys.path.append(EOS_LIB_DIRECTORY)
+from eos_transaction_builder import Transaction
+from eos import EosClient
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--path', help="BIP 32 path to use")
+parser.add_argument('--file', help="Transaction in JSON format")
+args = parser.parse_args()
+
+if args.path is None:
+    args.path = "m/44'/194'/0'/0/0"
+
+if args.file is None:
+    args.file = 'corpus/transaction.json'
+
+eos_path = pack_derivation_path(args.path)
+
+with open(args.file) as f:
+    obj = json.load(f)
+signing_digest, message = Transaction().encode(obj)
+
+with LedgerCommBackend(None, interface="hid") as backend:
```

Comment:
It's the remaining of the previously existing tests.
They allow to easily send a transaction in JSON format to real devices (and now speculos too).
This might comes handy to test some device behavior for some transactions.